core/Plugin/Visualization.php

```diff
@@ -458,13 +457,34 @@ private function postDataTableLoadedFromAPI()
         }
 
         // deal w/ table metadata
+        $metadata = null;
         if ($this->dataTable instanceof DataTable) {
-            $this->metadata = $this->dataTable->getAllTableMetadata();
+            $metadata = $this->dataTable->getAllTableMetadata();
 
-            if (isset($this->metadata[DataTable::ARCHIVED_DATE_METADATA_NAME])) {
-                $this->reportLastUpdatedMessage = $this->makePrettyArchivedOnText();
+
+        } else {
+            // if the dataTable is Map
+            if ($this->dataTable instanceof DataTable\Map) {
+                // load all the data
+                $dataTable = $this->dataTable->getDataTables();
+                // find the latest key
+                foreach ($dataTable as $item) {
+                    $metaData = $item->getAllTableMetadata();
+                    // if ts_created not exist exit loop
+                    if (empty($metaData[DataTable::ARCHIVED_DATE_METADATA_NAME])) {
+                        continue;
+                    }
+                    // if ts_created not record in the metadata or if current is more recent up the metadata
+                    if (!empty($metadata[DataTable::ARCHIVED_DATE_METADATA_NAME])
+                      || strtotime($metaData[DataTable::ARCHIVED_DATE_METADATA_NAME]) > strtotime($metadata[DataTable::ARCHIVED_DATE_METADATA_NAME])) {
+                        $metadata = $metaData;
```

Comment:
Don't you think it would be a bit easier to read by renaming one variable. When quickly looking over the code it might be hard to see that those variables differ in their writing case. This might also be a bit error prone when the code needs to be changed.